Allow user to actually set UTF-8 as the fallback charset.

This is to allow UTF-8 as default for text/plain

--- ./dom/encoding/FallbackEncoding.cpp.orig
+++ ./dom/encoding/FallbackEncoding.cpp
@@ -55,8 +55,7 @@
   // Don't let the user break things by setting the override to unreasonable
   // values via about:config
   if (!EncodingUtils::FindEncodingForLabel(override, mFallback) ||
-      !EncodingUtils::IsAsciiCompatible(mFallback) ||
-      mFallback.EqualsLiteral("UTF-8")) {
+      !EncodingUtils::IsAsciiCompatible(mFallback)) {
     mFallback.Truncate();
   }
 
